nixos/modules/system/boot/systemd.nix

```diff
@@ -35,7 +35,7 @@ let
       "nss-lookup.target"
       "nss-user-lookup.target"
       "time-sync.target"
-    ] ++ (optionals cfg.package.withCryptsetup [
+    ] ++ (optionals systemd.withCryptsetup [
```

Comment:
Shouldn't this all stay `cfg.package`, if you set `systemd.package` to something with/without a feature enabled?

I'd assume we only want to add the cryptsetup unit files to our system if the systemd we configured the system with has cryptsetup support enabled.

Reply:
https://github.com/NixOS/nixpkgs/blob/1118c797cf4fbb184030d4190e2e3eb42df67fc4/nixos/modules/system/boot/systemd.nix#L11

`cfg.package` and `systemd` are the same, or am I missing something?

Reply:
Nope you're right. Think @flokli was confused with `pkgs.systemd`

Reply:
Huh, that's right. Maybe we can just use `cfg.package` consistently. There's just too many places in nixpkgs with a big fat `with pkgs;` - by using `cfg.package` it'd be more explicit.

Reply:
This sounds reasonable, I'll update the commits.